tiledb/api/c_api/config/config_api.cc

```diff
@@ -172,122 +172,77 @@ capi_return_t tiledb_config_iter_done(
  * Possible errors: Both `lhs` and `rhs` may be null or an invalid handle.
  * `equal` may be a null pointer
  */
-capi_return_t tiledb_config_compare(
-    tiledb_config_t* lhs, tiledb_config_t* rhs, uint8_t* equal) noexcept {
-  return tiledb::api::api_entry_plain<tiledb::api::tiledb_config_compare>(
-      lhs, rhs, equal);
-}
+CAPI_PLAIN_BEGIN(
+    config_compare, tiledb_config_t* lhs, tiledb_config_t* rhs, uint8_t* equal)
+CAPI_PLAIN_END(config_compare, lhs, rhs, equal)
 
-capi_return_t tiledb_config_iter_alloc(
+CAPI_ERROR_BEGIN(
+    config_iter_alloc,
     tiledb_config_t* config,
     const char* prefix,
-    tiledb_config_iter_t** config_iter,
-    tiledb_error_t** error) noexcept {
-  return api_entry_error<tiledb::api::tiledb_config_iter_alloc>(
-      error, config, prefix, config_iter);
-}
+    tiledb_config_iter_t** config_iter)
+CAPI_ERROR_END(config_iter_alloc, config, prefix, config_iter)
 
-capi_return_t tiledb_config_iter_reset(
+CAPI_ERROR_BEGIN(
+    config_iter_reset,
     tiledb_config_t* config,
     tiledb_config_iter_t* config_iter,
-    const char* prefix,
-    tiledb_error_t** error) noexcept {
-  return api_entry_error<tiledb::api::tiledb_config_iter_reset>(
-      error, config, config_iter, prefix);
-}
+    const char* prefix)
+CAPI_ERROR_END(config_iter_reset, config, config_iter, prefix)
 
 /*
  * API Audit: Void return means no possible signal for an error. No channel that
  * can return an error.
  * Possible errors: `config` may be null or an invalid handle.
  */
-void tiledb_config_iter_free(tiledb_config_iter_t** config_iter) noexcept {
-  tiledb::api::api_entry_void<tiledb::api::tiledb_config_iter_free>(
-      config_iter);
-}
+CAPI_VOID_BEGIN(config_iter_free, tiledb_config_iter_t** config_iter)
+CAPI_VOID_END(config_iter_free, config_iter)
 
-capi_return_t tiledb_config_iter_here(
+CAPI_ERROR_BEGIN(
+    config_iter_here,
     tiledb_config_iter_t* config_iter,
     const char** param,
-    const char** value,
-    tiledb_error_t** error) noexcept {
-  return api_entry_error<tiledb::api::tiledb_config_iter_here>(
-      error, config_iter, param, value);
-}
+    const char** value)
+CAPI_ERROR_END(config_iter_here, config_iter, param, value)
 
-capi_return_t tiledb_config_iter_next(
-    tiledb_config_iter_t* config_iter, tiledb_error_t** error) noexcept {
-  return api_entry_error<tiledb::api::tiledb_config_iter_next>(
-      error, config_iter);
-}
+CAPI_ERROR_BEGIN(config_iter_next, tiledb_config_iter_t* config_iter)
+CAPI_ERROR_END(config_iter_next, config_iter)
 
-capi_return_t tiledb_config_iter_done(
-    tiledb_config_iter_t* config_iter,
-    int32_t* done,
-    tiledb_error_t** error) noexcept {
-  return api_entry_error<tiledb::api::tiledb_config_iter_done>(
-      error, config_iter, done);
-}
\ No newline at end of file
+CAPI_ERROR_BEGIN(
+    config_iter_done, tiledb_config_iter_t* config_iter, int32_t* done)
+CAPI_ERROR_END(config_iter_done, config_iter, done)
```

Comment:
Doing all of this with the preprocessor is probably not my favorite thing -- I would normally use overloading with tags or some kind of metaprogramming technique.  However, IIUC, we can't do that here because we need C linkage, so another approach isn't immediately obvious.  Sometimes you just have to stamp out literal text.